common/json-schema-to-grammar.cpp

```diff
@@ -78,30 +47,24 @@ struct BuiltinRule {
     std::vector<std::string> deps;
 };
 
-const std::string _up_to_15_digits = build_repetition("[0-9]", 0, 15);
-
 std::unordered_map<std::string, BuiltinRule> PRIMITIVE_RULES = {
     {"boolean", {"(\"true\" | \"false\") space", {}}},
-    {"decimal-part", {"[0-9] " + _up_to_15_digits, {}}},
-    {"integral-part", {"[0-9] | [1-9] " + _up_to_15_digits, {}}},
+    {"decimal-part", {"[0-9]{1,16}", {}}},
+    {"integral-part", {"[0] | [1-9] [0-9]{1,15}", {}}},
```

Comment:
```suggestion
    {"integral-part", {"[0] | [1-9] [0-9]{0,15}", {}}},
```
Needs to be `{0,15}` in this variant instead of `{1,15}`, otherwise we can't have single-digit numbers like "1" as a valid number.